src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/BackflowConflictResolver.cs

```diff
@@ -199,49 +166,59 @@ await repo.CommitAsync(
-        var result = await repo.RunGitCommandAsync(["diff", "--name-only", "--diff-filter=U", "--relative"], cancellationToken);
-        if (!result.Succeeded)
+        foreach (var conflictedFile in conflictedFiles)
         {
-            await AbortMerge();
-            result.ThrowIfFailed("Failed to resolve version file conflicts - failed to get a list of conflicted files");
-        }
-
-        var conflictedFiles = result.StandardOutput
-            .Split(Environment.NewLine, StringSplitOptions.RemoveEmptyEntries)
-            .Select(line => line.Trim())
-            .ToList();
+            // Known version file - check out the branch version, we want to override it
+            // See https://github.com/dotnet/arcade-services/issues/4865
+            if (DependencyFileManager.DependencyFiles.Any(f => f.Equals(conflictedFile, StringComparison.OrdinalIgnoreCase)))
+            {
+                // Revert files so that we can resolve the conflicts
+                // We use the target branch version when we are flowing the first time (because we did not flow the version files yet)
+                // We use the head branch version when we are flowing again because it already has updates from previous flow
+                // plus it can contain additional changes from the PR
+                await repo.RunGitCommandAsync(["checkout", headBranchExisted ? "--ours" : "--theirs", conflictedFile], cancellationToken);
+                continue;
+            }
 
-        var unresolvableConflicts = conflictedFiles
-            .Except(DependencyFileManager.DependencyFiles)
-            .ToList();
+            // Unknown conflict, but can be conflicting with a out-of-order recent flow
+            // Check DetectRecentFlow documentation for more details
+            if (recentFlow != null)
+            {
+                if (await TryResolvingConflictUsingRecentFlow(
+                    mapping.Name,
+                    vmr,
+                    repo,
+                    conflictedFile,
+                    currentFlow,
+                    recentFlow,
+                    cancellationToken))
+                {
+                    continue;
+                }
+            }
 
-        if (unresolvableConflicts.Count > 0)
-        {
-            _logger.LogInformation("Failed to merge the branch {targetBranch} into {headBranch} due to unresolvable conflicts: {conflicts}",
+            _logger.LogInformation("Failed to merge the branch {targetBranch} into {headBranch} due to unresolvable conflict in {conflictedFile}",
                 branchToMerge,
-                headBranch,
-                string.Join(", ", unresolvableConflicts));
+                targetBranch,
+                conflictedFile);
 
-            await AbortMerge();
+            await AbortMerge(repo);
 
             // If this is a first commit, we need to update dependencies still
```

Comment:
I'm trying to figure out why we do this at all, is it so we have changes to open a PR even if we have conflicts?

Reply:
I think you might have discovered a problem of the refactoring. I made everything simpler - we try to always merge (or merge + resolve conflicts) and regardless of how that went, we update dependencies.